language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -107,31 +111,75 @@ genModule curPkgId mod
     in
     Just $ T.unlines $ intercalate [""] $ filter (not . null) $ header : imports : defs
   where
-    serDefs = filter (getIsSerializable . dataSerializable) (NM.toList (moduleDataTypes mod))
+    serDefs = defDataTypes mod
+
+defDataTypes :: Module -> [DefDataType]
+defDataTypes mod = filter (getIsSerializable . dataSerializable) (NM.toList (moduleDataTypes mod))
 
-genDataDef :: PackageId -> ModuleName -> NM.NameMap Template -> DefDataType -> (([T.Text], [T.Text]), Set.Set ModuleRef)
-genDataDef curPkgId curModName tpls def = case unTypeConName (dataTypeCon def) of
+genDataDef :: PackageId -> Module -> NM.NameMap Template -> DefDataType -> (([T.Text], [T.Text]), Set.Set ModuleRef)
+genDataDef curPkgId mod tpls def = case unTypeConName (dataTypeCon def) of
     [] -> error "IMPOSSIBLE: empty type constructor name"
     _: _: _: _ -> error "IMPOSSIBLE: multi-part type constructor of more than two names"
-    [conName] -> genDefDataType curPkgId conName curModName tpls def
-    [c1, c2] -> ((makeNamespace $ map ("  " <>) typs, makeNamespace $ map ("  " <>) sers), refs)
+
+    [conName] -> genDefDataType curPkgId conName mod tpls def
+    [c1, c2] -> ((makeNamespace $ map ("  " <>) typs, []), refs)
       where
-        ((typs, sers), refs) = genDefDataType curPkgId c2 curModName tpls def
-        ns = c1 <> "_NS"
+        ((typs, _), refs) = genDefDataType curPkgId c2 mod tpls def
+        ns = c1
```

Comment:
Why not inline this?

Reply:
No deep reason - legacy I'd say. I'll have another look.